releasenotes/notes/42173.yaml

```diff
@@ -0,0 +1,12 @@
+apiVersion: release-notes/v2
+kind: bug-fix
+area: traffic-management
+
+# issue is a list of GitHub issues resolved in this note.
+issue:
+  - https://github.com/istio/istio/issues/42173
+  - 42173
+
+releaseNotes:
+  - |
+    **Fixed** bug when set `ENABLE_ENHANCED_RESOURCE_SCOPING=true`, istio and k8s gateway api resources will not be handled correctly when namespace selected or deselected by discovery selector or namespace label change.
```

Comment:
nit: or namespace label change -> or namespace label is changed

Reply:
ok